src/portal-impl.c

```diff
@@ -551,19 +551,29 @@ find_portal_implementation (const char *interface)
 GPtrArray *
 find_all_portal_implementations (const char *interface)
 {
+  const char **desktops;
   GPtrArray *impls;
   GList *l;
+  int i;
 
   impls = g_ptr_array_new ();
 
-  for (l = implementations; l != NULL; l = l->next)
-    {
-      PortalImplementation *impl = l->data;
+  desktops = get_current_lowercase_desktops ();
 
-      if (g_strv_contains ((const char **)impl->interfaces, interface))
+  for (i = 0; desktops[i] != NULL; i++)
+    {
```

Comment:
I don't see desktops[i] being used anywhere in the body of the loop. What am I missing?

Reply:
Ah, I accidentally copied dead code from `find_portal_implementation` that was left behind when the UseIn= fallback was refactored to its own loop.

Fixed, and added a commit that removes the dead code at the source while at it.